php-frontend/src/main/java/org/sonar/php/tree/symbols/SymbolVisitor.java

```diff
@@ -551,29 +549,50 @@ private void usageForNamespaceName(NamespaceNameTree namespaceName, Symbol.Kind
 
   @Override
   public void visitMemberAccess(MemberAccessTree tree) {
-    memberStatementDepth++;
-    if (memberStatementDepth >= MEMBER_ACCESS_MAXIMUM_DEPTH) {
-      if (LOG.isDebugEnabled()) {
-        LOG.debug("visitMemberAccess at \"{}...\" has reached the maximum depth of {}, switching to the next statement",
-          tree.toString().substring(0, MEMBER_ACCESS_LOG_TRUNCATION_LENGTH), MEMBER_ACCESS_MAXIMUM_DEPTH);
-      }
-      return;
```

Comment:
Aborting the construction of the symbol table doesn't work as I expected; this breaks our assumptions about it. Instead, for this particular case, I'm using a loop instead of recursion. The test with 1500 chained calls throws StackOverflow on master but passes with this change.